android/java/org/chromium/chrome/browser/crypto_wallet/activities/AssetDetailActivity.java

```diff
@@ -444,13 +444,17 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
     }
 
     private void adjustButtonsVisibilities() {
-        if (Utils.allowBuyAndSwap(mChainId)) {
+        if (Utils.allowBuyOrSwap(mChainId, BuySendSwapActivity.ActivityType.BUY)) {
+            mBtnBuy.setVisibility(View.VISIBLE);
+        } else {
+            mBtnBuy.setVisibility(View.GONE);
+        }
+        if (Utils.allowBuyOrSwap(mChainId, BuySendSwapActivity.ActivityType.SWAP)) {
             mBtnBuy.setVisibility(View.VISIBLE);
```

Comment:
do we need to make buy visible for swap condition? not required?

Reply:
It was an oversight. Fixed now.